### PR TITLE
Moved troubleshooting from build.md to its own document.

### PR DESCRIPTION
**Pull request by evildandelions**
Moving troubleshooting into its own document. Added mouseover links for each item. 

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1598)

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## Troubleshooting 
+
+##### Make sure `$GOPATH` is set, and your [workspace](https://golang.org/doc/code.html#Workspaces) is configured.
+
+##### kops will not compile with symlinks in `$GOPATH`. See issue go issue [17451](https://github.com/golang/go/issues/17451) for more information
+
+##### kops uses the relatively new Go vendoring, so building requires Go 1.6 or later, or you must export GO15VENDOREXPERIMENT=1 when building with Go 1.5.  The makefile sets GO15VENDOREXPERIMENT for you.  Go code generation does not honor the env var in 1.5, so for development you should use Go 1.6 or later
```
Comment (chrislovecnm) [inline docs/troubleshooting.md]: We do not support 1.5 anymore :)  You want to remove the entire section about go 1.5?  We only support 1.6 and 1.7.

Comment (evildandelions) [inline docs/troubleshooting.md]: Hey @chrislovecnm I fixed the references to Go 1.5. Thanks for the catch. Is there anything else you'd suggest for this PR?

Comment (evildandelions) [inline docs/troubleshooting.md]: Also I updated my description for the PR, sorry about it being blank. 

Do you think we should point to this from `README.md` ?

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+## Troubleshooting
```
Comment (chrislovecnm) [inline docs/troubleshooting.md]: Looking at this again ... This is a pretty generic title.  Should we have a section for troubleshooting the build?

Comment (chrislovecnm) [inline docs/troubleshooting.md]: Appreciate your patience with the nitpick, and I should have mentioned it earlier.

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

File: docs/troubleshooting.md (hunk 2 above)
Comment (evildandelions) [inline docs/troubleshooting.md]: I changed the title, can you take a look at it?

File: docs/install.md [hunk 3]
```diff
@@ -24,43 +24,3 @@ make
...
 Cross compiling for things like `nodeup` are now done automatically via `make nodeup`. `make push-aws-run TARGET=admin@$TARGET` will automatically choose the linux amd64 build from your `.build` directory.
 
-## Troubleshooting 
-
- - Make sure `$GOPATH` is set, and your [workspace](https://golang.org/doc/code.html#Workspaces) is configured.
- - kops will not compile with symlinks in `$GOPATH`. See issue go issue [17451](https://github.com/golang/go/issues/17451) for more information
- - kops uses the relatively new Go vendoring, so building requires Go 1.6 or later, or you must export GO15VENDOREXPERIMENT=1 when building with Go 1.5.  The makefile sets GO15VENDOREXPERIMENT for you.  Go code generation does not honor the env var in 1.5, so for development you should use Go 1.6 or later
- - Kops will only compile if the source is checked out in `$GOPATH/src/k8s.io/kops`. If you try to use `$GOPATH/src/github.com/kubernetes/kops` you will run into issues with package imports not working as expected.
-
-# Installing Other Dependencies
```
Comment (justinsb) [inline docs/install.md]: Did we remove from here on deliberately?

File: docs/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Troubleshooting
```
Comment (justinsb) [inline docs/troubleshooting.md]: This is troubleshooting building from source, so I'm not sure that it is right to split it off from the building-from-source instructions.  Thoughts @evildandelions ?

Comment (evildandelions) [inline docs/troubleshooting.md]: @justinsb I was thinking that all troubleshooting topics would eventually find their way into the same document.

**Comment by justinsb** [comment]
So two things and then we can merge:

* I don't think we want to lose the additional non-troubleshooting instructions from the first file
* Can we rename docs/troubleshooting.md to docs/development/troubleshooting.md or something else developer focused?  I do want (1) to have a troubleshooting doc for end users and (2) to try to dissuade end users from building from source

Thanks for bearing with us @evildandelions  and let's get this merged :-)

**Comment by chrislovecnm** [comment]
Bump

**Comment by chrislovecnm** [comment]
Can we close this?

**Comment by justinsb** [comment]
I'm just going to close; please feel free to bump if you want to pursue @evildandelions - would be great if so!

**Pull request closed without merging**